cegs_portal/search/views/custom_views.py

```diff
@@ -1,20 +1,51 @@
+import csv
+import io
 import logging
 from typing import Any, Callable, Optional
 
 from django.contrib.auth.mixins import UserPassesTestMixin
 from django.core.exceptions import BadRequest
-from django.http.response import JsonResponse
+from django.http.response import HttpResponse, JsonResponse
 from django.shortcuts import render
 from django.views.generic import View
 
 from cegs_portal.search.views.renderers import json
-from cegs_portal.search.views.view_utils import JSON_MIME
+from cegs_portal.search.views.view_utils import JSON_MIME, TSV_MIME
 from cegs_portal.utils.http_exceptions import Http303, Http400, Http500
 from cegs_portal.utils.http_responses import HttpResponseSeeOtherRedirect
 
 logger = logging.getLogger("django.request")
 
 
+class TsvResponse(HttpResponse):
+    """
+    An HTTP response class that consumes data to be serialized to JSON.
+
+    :param data: Data to be dumped into json. By default only ``dict`` objects
+      are allowed to be passed due to a security flaw before ECMAScript 5. See
+      the ``safe`` parameter for more information.
+    :param encoder: Should be a json encoder class. Defaults to
+      ``django.core.serializers.json.DjangoJSONEncoder``.
+    :param safe: Controls if only ``dict`` objects may be serialized. Defaults
+      to ``True``.
+    :param json_dumps_params: A dictionary of kwargs passed to json.dumps().
+    """
+
+    def __init__(
+        self,
+        data,
+        **kwargs,
+    ):
+        kwargs.setdefault("content_type", "text/tab-separated-values")
```

Comment:
in addition to setting the content type you should also set the content disposition, like in https://github.com/ReddyLab/cegs-portal/blob/ce3341995c330dd2060115980902db58c8f91e80/cegs_portal/search/views/v1/non_targeting_reos.py#L107 which means you'll also need a way to set the file name.